### ISSUE TITLE
How to disable parallelism for a test collection?

### ISSUE THREAD

**Issue by thomaslevesque** [xunit/xunit#1179]
I have a collection of unit tests that can't run in parallel, because they mutate global state and interfere with each other. I can put them in a collection so that they're not run in parallel against each other, but they can also interfere with the rest of the tests (those which don't mutate global state).

Is there a way to also prevent *any other test* from running at the same time at this collection? To be clear, I want this collection to run alone, and the rest of the tests to run in parallel.

Thanks

**Comment by cwienands1** [comment]
Could you split your test project into two? Put the tests that don't play nice into a separate assembly and use assembly-level setting parallelizeAssembly=false on that one to avoid that it gets executed in parallel with the original test assembly (or others).

**Comment by thomaslevesque** [comment]
@cwienands1 I guess I could do that. IIRC each assembly is executed in its own AppDomain anyway, so it will be run in isolation anyway.

**Comment by bradwilson** [comment]
1. App domains only apply to the desktop CLR
2. There are some process-global things that are not isolated per app-domain (for example, "current directory").

**Comment by thomaslevesque** [comment]
@bradwilson good points.

So, is there any way to achieve what I described? Besides creating a separate project?

**Comment by bradwilson** [comment]
Nope. Move it to a separate assembly, and run that assembly separately from the rest of the assemblies.

**Comment by thomaslevesque** [comment]
OK, thanks

**Comment by otto-gebb** [comment]
@thomaslevesque It seems to be possible in 2.3.0, see the [DisableParallelization](https://github.com/xunit/xunit/blob/3d01cb1a69f0bd18558abf3fa6e6ad72b84ea174/test/test.xunit.execution/Acceptance/Xunit2AcceptanceTests.cs#L391) property.

**Comment by thomaslevesque** [comment]
@otto-gebb thanks, I'll give it a try. @bradwilson can you confirm this is what I'm looking for? From the doc comments, it seems to be what I need, but I might have misunderstood.

**Comment by bradwilson** [comment]
This feature was added in September in [2.3 beta 5](https://xunit.github.io/releases/2.3-beta5):

> Added the ability to disable cross-collection parallelization for individual test collections, via the test collection definition (by setting `[CollectionDefinition(DisableParallelization = true)]`). Parallel-capable test collections will be run first (in parallel), followed by parallel-disabled test collections (run sequentially).

I believe this should solve your problem.

**Comment by thomaslevesque** [comment]
Perfect, thanks!

**Comment by elladan06** [comment]
I tried the attribute and my tests still run in parallel. 

I added this attribute on the two class I want to disable de parallelization.
`[CollectionDefinition("EndpointTests", DisableParallelization = true)]`

I'm using the XUnit and XUnit.runner.visualstudio at version 2.4.1.

To fix this I'm using a AutoResetEvent to ensure only one test is ran at the time.

**Comment by bradwilson** [comment]
@elladan06 This is a closed issue. Please open a new issue with a repro project.
